### PR TITLE
refactor(amt_libraries): updated folder structure. added functionality

### PR DESCRIPTION
**Pull request by matt-primrose**
modularized folder structure
moved interfaces and tests into amt folder
moved repetitive functions into generic private functions:
- Get
- Pull
- Enumerate
- Put
- Delete
- RequestStateChange

split out enums to allow for better reuse
added error handling
added WSMAN class support for:
- AMT_AuditLog
- AMT_RedirectionService
- AMT_SetupAndConfigurationService
- AMT_GeneralSettings
- AMT_EthernetPortSettings
- AMT_RemoteAccessPolicyRule
- AMT_ManagementPresenceRemoteSAP
- AMT_PublicKeyCertificate
- AMT_EnvironmentDetectionSettingData
- AMT_PublicKeyManagementService
- AMT_RemoteAccessService
- AMT_UserInitiatedConnectionServic
- AMT_BootSettingData
- CIM_ComputerSystemPackage
- CIM_SoftwareIdentity
- CIM_SystemPackaging
- CIM_KVMRedirectionSAP
- CIM_Chassis
- CIM_Chip
- CIM_Card
- CIM_BIOSElement
- CIM_Processor
- CIM_PhysicalMemory
- CIM_MediaAccessDevice
- CIM_PhysicalPackage
- CIM_WiFiEndpointSettings
- CIM_BootService
- CIM_BootConfigSetting
- CIM_PowerManagementService
- IPS_HostBasedSetupService

### CODE REVIEW

**Review by rsdmike — CHANGES_REQUESTED** [review]
It is a nit, but anywhere you have `String`, we should be using `string`. It's a super small difference, the `String` class gives ya a bit extra stuff that we dont need, `string` is the javascript primitive type and is preferred.

File: src/amt/cim.ts [hunk 1]
```diff
@@ -3,56 +3,251 @@
...
+  }
+
+  cim_SoftwareIdentity = (method: CIM_Methods, messageId: String, enumerationContext?: String): String => {
+    switch (method) {
+      case 'Enumerate':
+        return this.enumerate(CIM_Actions.ENUMERATE, CIM_Classes.CIM_SOFTWARE_IDENTITY, messageId)
+      case 'Pull':
+        if (enumerationContext != null) {
+          return this.pull(CIM_Actions.PULL, CIM_Classes.CIM_SOFTWARE_IDENTITY, messageId, enumerationContext)
+        } else {
```
Comment (rsdmike) [inline src/amt/cim.ts:51]: else is unnecessary, could even do a fallthrough statement to execute the default: but some people don't prefer that.

File: src/amt/cim.ts [hunk 2]
```diff
@@ -3,56 +3,251 @@
...
+          return null
+        }
+      default:
+        return null
+    }
+  }
+
+  cim_Card = (method: CIM_Methods, messageId: String): String => {
+    switch (method) {
```
Comment (rsdmike) [inline src/amt/cim.ts:119]: I know you're going for consistency with these, but with only one/two cases i'm not sure legibility is the greatest here -- are there more cases that are coming?

File: src/amt/cim.ts [hunk 3]
```diff
@@ -3,56 +3,251 @@
...
+    const header: String = this.wsmanMessageCreator.createHeader(action, `${this.resourceUriBase}${cimClass}`, messageId)
+    const body: String = this.wsmanMessageCreator.createBody(CIM_Methods.GET)
+    const response: String = this.wsmanMessageCreator.createXml(header, body)
+    return response
   }
 
   cim_ServiceAvailableToElement = (method: CIM_Methods, messageId: String, enumerationContext?: String): String => {
-    let response: String, header: String, body: String
+    switch (method) {
+      case 'Pull':
```
Comment (rsdmike) [inline src/amt/cim.ts:35]: why not use the constant here in all the case conditions?

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/amt/cim.ts (hunk 1 above)
Comment (matt-primrose) [reply src/amt/cim.ts:51]: updated

File: src/amt/cim.ts (hunk 2 above)
Comment (matt-primrose) [reply src/amt/cim.ts:119]: All of these potentially have up to 6 methods available (Get, Pull, Enumerate, Put, Subscribe, Unsubscribe) that come from the base CIM model.  Currently I'm just implementing the ones that we use.

File: src/amt/cim.ts (hunk 3 above)
Comment (matt-primrose) [reply src/amt/cim.ts:35]: agree.  There are a couple spots where the body is conditionally updated, but for everything else I'm using constants or just returning the response directly when it comes back from the called function.

**Review by rsdmike — APPROVED** [review]

**Pull request merged**